Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Registration.hs

```diff
@@ -139,6 +140,15 @@ login LoginReq {..} = do
       let merchantIds = map DAccess.merchantShortId $ map head groupedByMerchant
       merchants <- QMerchant.findAllByShortIds merchantIds
       let enabledMerchants = filter (\merchant -> merchant.enabled == Just True) merchants
+      forM_ enabledMerchants $ \merchant' ->
+        case merchant'.passwordExpiryDays of
+          Just days -> do
+            let updatedAt = person.updatedAt
+                secondsSinceUpdate = diffUTCTime now updatedAt
+                expiryLimit = fromIntegral days * 86400
+            when (secondsSinceUpdate > expiryLimit) $
+              throwError $ InvalidRequest "Your password has expired. Please check with admin."
```

Comment:
This way you check password expiry only when user try to login. But still user can call any api.

You can add password expiry check in next fucntions:
- `Tools.Auth.Api.verifyApi` - user access
- `Tools.Auth.Dashboard.verifyDashboard` - admin access

This way password expiry will be checked for every api call in middleware (`App.runService` func):

```
    context =
      Auth.verifyApiAction @(FlowR AppEnv)
        :. Auth.verifyDashboardAction @(FlowR AppEnv)
        :. EmptyContext
```